src/main/scala/lms/thirdparty/cuda/cudnn.scala

```diff
@@ -0,0 +1,407 @@
+package lms.thirdparty
+
+import lms.core._
+import lms.util._
+import lms.core.stub._
+import lms.core.Backend._
+import lms.core.virtualize
+import lms.core.utils.time
+import lms.macros.SourceContext
+import lms.thirdparty.array_computation.CudaOps
+
+object CUDNNTypeLess extends Dsl with CLibs {
+
+  import BaseTypeLess._
+  import PrimitiveTypeLess._
+  import SIZE_TTypeLess._
+  import CLibTypeLess._
+
+  class CUDNN_RESULT
+
+  def CUDNN_CHECK(result: TOP)(implicit __pos: SourceContext): UNIT = {
+    assert(result.t == manifest[CUDNN_RESULT], "CUDNN_CHECK must take the CUDNN_RESULT type as input")
+    UNIT(LIB_FUNCTION(manifest[Unit], "CUDNNCHECK", result.x)(Seq[Int](), Seq[Int](), Set[Int](), Adapter.CTRL))
+  }
+} 
+
+
+trait CUDNNOps extends CLibs with CudaOps {
+
+  /* LMS support for CUDNN library */
+  
+  class cudnnStatusT
+
+  class cudnnHandleT
+  def cudnnHandle: Rep[cudnnHandleT] = newStruct[cudnnHandleT]("cudnnHandle_t")
```

Comment:
we might also want to provide a `cached` cudnnHandle, so that one cudnnHandle is used throughout.

You can check how cublasHandle is done :)

Reply:
cached cudnnHandle added to cudnn.scaka